isic/core/search.py

```diff
@@ -0,0 +1,167 @@
+                {'from': 0.5, 'to': 1.0},
+                {'from': 1.0, 'to': 1.5},
+                {'from': 1.5, 'to': 2.0},
+                {'from': 2.0, 'to': 2.5},
+                {'from': 2.5, 'to': 3.0},
+                {'from': 3.0, 'to': 3.5},
+                {'from': 3.5, 'to': 4.0},
+                {'from': 4.0, 'to': 4.5},
+                {'from': 4.5, 'to': 5.0},
+                {'from': 5.0},
+            ],
+        }
+    },
+    'mel_type': {'terms': {'field': 'mel_type'}},
+    'mel_ulcer': {'terms': {'field': 'mel_ulcer'}},
+}
+
+
+@lru_cache
+def get_elasticsearch_client():
+    return Elasticsearch(settings.ISIC_ELASTICSEARCH_URI)
+
+
+def maybe_create_index():
+    # TODO: include private meta fields (e.g. patient/lesion id)
+    get_elasticsearch_client().indices.create(
+        index=settings.ISIC_ELASTICSEARCH_INDEX,
+        ignore=400,  # ignore status code 400 (index already exists)
+        body={
+            'mappings': {
+                'properties': {
+                    'created': {'type': 'date'},
+                    # TODO: unique index name?
+                    'name': {'type': 'text'},
+                    'public': {'type': 'boolean'},
+                    'age_approx': {'type': 'integer'},
+                    'sex': {'type': 'keyword'},
+                    'benign_malignant': {'type': 'keyword'},
+                    'diagnosis': {'type': 'keyword'},
+                    'diagnosis_confirm_type': {'type': 'keyword'},
+                    'personal_hx_mm': {'type': 'boolean'},
+                    'family_hx_mm': {'type': 'boolean'},
+                    'clin_size_long_diam_mm': {'type': 'float'},
+                    'melanocytic': {'type': 'boolean'},
+                    'acquisition_day': {'type': 'float'},
+                    'marker_pen': {'type': 'boolean'},
+                    'hairy': {'type': 'boolean'},
+                    'blurry': {'type': 'boolean'},
+                    'nevus_type': {'type': 'keyword'},
+                    'image_type': {'type': 'keyword'},
+                    'dermoscopic_type': {'type': 'keyword'},
+                    'anatom_site_general': {'type': 'keyword'},
+                    'color_tint': {'type': 'keyword'},
+                    'mel_class': {'type': 'keyword'},
+                    'mel_mitotic_index': {'type': 'keyword'},
+                    'mel_thick_mm': {'type': 'float'},
+                    'mel_type': {'type': 'keyword'},
+                    'mel_ulcer': {'type': 'boolean'},
+                }
+            }
+        },
+    )
+
+
+def add_to_search_index(image) -> None:
+    get_elasticsearch_client().index(
+        index=settings.ISIC_ELASTICSEARCH_INDEX, body=image.as_elasticsearch_document
+    )
+
+
+def bulk_add_to_search_index(qs: QuerySet) -> None:
+    def image_documents():
+        for image in qs.select_related('accession').all():
+            yield image.as_elasticsearch_document
```

Comment:
```suggestion
    image_documents = (
    	image.as_elasticsearch_document
    	for image in qs.select_related('accession').all()
  	)
```
Could this just be anything iterable?